### ISSUE TITLE
Fuelux 3.11.0 not installable on Windows

### ISSUE THREAD

**Issue by maoueh** [ExactTarget/fuelux#1444]
Hi there,

`Fuelux` 3.11.0 is not installable on Windows because of the usage of `./postinstall.sh` as the post-install script. Npm complains that `.` is not a valid command or external program. Using `sh postinstall.sh` would work on my machine (because I use a Unix emulation layer `msys2` to not name it). But it would not work on all Windows machine.

It's not yet available on `npm` registry but installing via github does the same as `npm` registry I guess. I tried `3.11.0` that fast because we had problem installing previous `npm` released version (`3.10.0`) since `bower` is in `devDependencies` and as such, `postinstall` script was not working.

Here test `package.json`:

 ```
{
  "name": "fuelux-install-test-windows",
  "version": "0.0.1",
  "private": "true",
  "dependencies": {
    "fuelux": "ExactTarget/fuelux.git#3.11.0"
  }
}
```

And `npm install` output:

```
$ npm install
npm WARN engine fuelux@3.11.0: wanted: {"node":"0.10.x"} (current: {"node":"0.12.2","npm":"2.7.4"})

> fuelux@3.11.0 postinstall C:\Users\movachon\Scrapbook\web\fuelux-install\node_modules\fuelux
> ./postinstall.sh

'.' is not recognized as an internal or external command,
operable program or batch file.
npm ERR! Windows_NT 6.1.7601
npm ERR! argv "C:\\Local\\Node\\0.12.2\\node.exe" "C:\\Local\\Node\\0.12.2\\node_modules\\npm\\bin\\npm-cli.js" "install"
npm ERR! node v0.12.2
npm ERR! npm  v2.7.4
npm ERR! code ELIFECYCLE

npm ERR! fuelux@3.11.0 postinstall: `./postinstall.sh`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the fuelux@3.11.0 postinstall script './postinstall.sh'.
npm ERR! This is most likely a problem with the fuelux package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     ./postinstall.sh
npm ERR! You can get their info via:
npm ERR!     npm owner ls fuelux
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\movachon\Scrapbook\web\fuelux-install\npm-debug.log
```

Regards,
Matt

**Comment by swilliamset** [comment]
My apologies to our Windows user friends. We will certainly get this cleaned up.

**Comment by maoueh** [comment]
No problem. I'm willing to propose a patch but I would like to understand the intention. Here my understanding right now.

 * Prior say 3.10.0 (did not search extensively which version exactly), `bower` was a core dependencies and was listed in `dependencies` bucket of the `package.json`.
 * After 3.10.0, `bower` is now a development dependencies and as such, previous post install script could not work out of the box anymore. So you [removed it](https://github.com/ExactTarget/fuelux/pull/1402).
 *  Then it was [re-added as a bash script](https://github.com/ExactTarget/fuelux/pull/1404) but conditionally with an environment variable.

Was the last step to ease developer working on `fuelux`? For easier `CI` building? What would be the impact of simply dropping post-install script?

**Comment by interactivellama** [comment]
We auto-deploy Fuel UX to Heroku for designers and others to be able to view the current state of #master branch visually and interact with it.

Now that Fuel UX can be installed on NPM, bower is still needed for the deploy. This is an ENV variable flag to ask Heroku to install Bower, but only Heroku or other deploys that need it. For Fuel UX to deploy as an app itself we can just ask that it do a development install with `devDependencies`, but in apps that need Fuel UX as an NPM dependency, it is being asked for the production version of Fuel UX and therefore does not install Bower. An example: https://github.com/ExactTarget/fuelux-mctheme/blob/master/package.json#L30

In summary, it's a hack to make Bower work in apps that have Fuel UX as an NPM dependency and use bower in the front end `<script>` tags.

Another answer and probably the best one is to move these "testing/development/edge version apps" away from Bower usage on the front end and make them use NPM scripts from the `node_modules` folder. These are not "production apps," so they do not need to bundle their dependencies, nor hide `node_modules`.

In the mean time, do you have a suggestion for optional install scripts that work in Windows and *nix for the `postinstall` key?

**Comment by maoueh** [comment]
Turning it into a `node` script is probably what I would suggest.

Like https://github.com/ExactTarget/fuelux/compare/master...maoueh:fix-windows-npm-install

I use the bower programmatic API but we could launch an command process instead. Tested on my workstation, worked great. Do you want me to open a PR?

**Comment by interactivellama** [comment]
Sounds great. Please open.

**Comment by swilliamset** [comment]
Collaboration!!! :+1:

**Comment by maoueh** [comment]
Closing since fix has been merged. Thank you all.
